### PR TITLE
Feature/issue 03

### PR DESCRIPTION
**Pull request by kevin-verschaeve**
https://github.com/poledev/kevin/issues/9

### CODE REVIEW

File: app/config/config.yml [hunk 1]
```diff
@@ -72,3 +73,10 @@ swiftmailer:
     #scope:          %sensiolabs_connect.scope%
     #oauth_endpoint: %sensiolabs_connect.oauth_endpoint%
     #api_endpoint:   %sensiolabs_connect.api_endpoint%
+
+stof_doctrine_extensions:
+    default_locale: en_EN
+    orm:
+        default:
+            timestampable: true
+            sluggable: true
```
Comment (jderusse) [inline app/config/config.yml]: missing end line

File: app/config/config.yml [hunk 2]
```diff
@@ -72,3 +73,10 @@ swiftmailer:
     #scope:          %sensiolabs_connect.scope%
     #oauth_endpoint: %sensiolabs_connect.oauth_endpoint%
     #api_endpoint:   %sensiolabs_connect.api_endpoint%
+
+stof_doctrine_extensions:
+    default_locale: en_EN
```
Comment (jderusse) [inline app/config/config.yml]: Use parameter `%locale%`

File: app/config/services.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+services:
+    sensiolabs.job_form:
+            class: SensioLabs\JobBoardBundle\Form\Type\JobType
+            arguments: [@router]
+            tags:
+                - { name: form.type, alias: jobboard_post }
```
Comment (jderusse) [inline app/config/services.yml]: missing end line

File: src/SensioLabs/JobBoardBundle/Controller/BaseController.php [hunk 4]
```diff
@@ -28,7 +30,41 @@ public function indexAction(Request $request)
...
+            $em->flush();
+
+            return $this->redirectToRoute('job_preview', array(
+                'countryCode' => $job->getCountry(),
+                'contractType' => $job->getContractType(),
+                'slug' => $job->getSlug()
+            ));
+        }
+
+        return array('form' => $form->createView());
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Controller/BaseController.php]: Doc say it returns a `RedirectResponse`

File: src/SensioLabs/JobBoardBundle/Controller/BaseController.php [hunk 5]
```diff
@@ -41,41 +41,60 @@ public function indexAction(Request $request)
...
         if ($form->isValid()) {
-            $this->saveJob($job);
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($job);
+            $em->flush();
 
             return $this->redirectToRoute('job_preview', array(
                 'countryCode' => $job->getCountry(),
                 'contractType' => $job->getContractType(),
-                'slug' => $job->getSlug(),
```
Comment (mmarchois) [inline src/SensioLabs/JobBoardBundle/Controller/BaseController.php]: Why remove this comma ?

File: src/SensioLabs/JobBoardBundle/Controller/BaseController.php [hunk 6]
```diff
@@ -41,41 +41,60 @@ public function indexAction(Request $request)
...
 
-        return ['form' => $form->createView()];
+        return array('form' => $form->createView());
     }
 
     /**
-     * @Route("/manage", name="manage")
-     * @Template()
+     * @Route("/post/{slug}", name="job_update")
+     * @Template("SensioLabsJobBoardBundle:Base:post.html.twig")
```
Comment (mmarchois) [inline src/SensioLabs/JobBoardBundle/Controller/BaseController.php]: IMO, you should use `render` method to render your template.
`@Template` annotation involve more magic.

File: src/SensioLabs/JobBoardBundle/Controller/BaseController.php [hunk 7]
```diff
@@ -41,41 +41,60 @@ public function indexAction(Request $request)
...
+     * @return array|\Symfony\Component\HttpFoundation\RedirectResponse
      */
-    public function manageAction()
+    public function updateAction(Job $job, Request $request)
     {
-        return [];
+        $form = $this->createForm('jobboard_post', $job);
+
+        $form->handleRequest($request);
+        if ($form->isValid()) {
```
Comment (mmarchois) [inline src/SensioLabs/JobBoardBundle/Controller/BaseController.php]: `if ($form-> handleRequest($request)->isValid())` ?

File: src/SensioLabs/JobBoardBundle/Controller/BaseController.php [hunk 8]
```diff
@@ -41,41 +41,60 @@ public function indexAction(Request $request)
...
     {
         $job = new Job();
-        $form = $this->createForm(new JobType(), $job);
+        $form = $this->createForm('jobboard_post', $job);
 
         $form->handleRequest($request);
         if ($form->isValid()) {
-            $this->saveJob($job);
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($job);
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Controller/BaseController.php]: Belongs to the model

File: src/SensioLabs/JobBoardBundle/Form/Type/JobType.php [hunk 9]
```diff
@@ -14,12 +15,27 @@
 class JobType extends AbstractType
 {
     /**
+     * @var RouterInterface $router
+     */
+    private $router;
```
Comment (mmarchois) [inline src/SensioLabs/JobBoardBundle/Form/Type/JobType.php]: /!\ Do not inject the Router component into a Form. Form action is based on your current route. Why do you need this ?

File: app/config/services.yml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+services:
+    sensiolabs.job_form:
+        class: SensioLabs\JobBoardBundle\Form\Type\JobType
+        arguments: [@router]
+        tags:
+            - { name: form.type, alias: jobboard_post }
```
Comment (Nicofuma) [inline app/config/services.yml]: This should be in your bundle services.[yml|xml]

File: app/config/config.yml [hunk 11]
```diff
@@ -72,3 +73,10 @@ swiftmailer:
     #scope:          %sensiolabs_connect.scope%
```
Comment (mmarchois) [inline app/config/config.yml]: Remove these comment lines

**Pull request merged**